src/DMS/Service/Meetup/Plugin/RateLimitPlugin.php

```diff
@@ -145,7 +145,10 @@ public function onBeforeSend()
      */
     protected function slowdownRequests()
     {
-        $microsecondsPerRequestRemaining = $this->rateLimitReset / $this->rateLimitRemaining * 1000000;
-        usleep($microsecondsPerRequestRemaining);
+        $sleepInMicroseconds = ((int) $this->rateLimitRemaining === 0)
```

Comment:
From a development / testing perspective I would advice to set a default value first, followed by a condition check to implement a conditional value. This way you reduce the Cyclomatic Complexity and apply a bail-first approach in your coding practices.

Also by casting the outcome immediately before storing it in `$sleepInMicroseconds` just keeps consistency and you know that the value will always be of type `int`. So you can use `usleep($sleepInMicroseconds)` directly making the code cleaner.

Something like the following:

```php
$sleepInMicroseconds = (int) $this->rateLimitReset * 1000000;
if (0 !== (int) $this->rateLimitRemaining) {
    $sleepInMicroseconds = (int) (($this->rateLimitReset / $this->rateLimitRemaining) * 1000000);
}
```

Just my 2-cents I advice to most of the developers I work with, love to hear your take on it.

Reply:
I beg to disagree in this case.

I would opt for early return/ bail out but the code construct here makes that approach more complex to human interpretation. I am a bug fan of not using else, but in this case the alternative carries much more code duplication that makes the reading experience more complex.

Same goes for casting here at store point, the ordering an syntax becomes rather weird, so just in time casting felt better, to be honest I'm rewriting this for PHP7 and will do strict types everywhere period, cause neither solution here is good IMO.

Reply:
I can agree with your point of view, that's why I love to have these discussions. 😺  Thank you for your feedback. I very much appreciate you pointing out the "unknown" fact that not everyone is used to this style of coding. Seems I need to contribute more so I can learn and become better at my trade.

Yes, PHP 7 helps a lot ensuring right types being used. Go PHP 7!